### PR TITLE
Fix classloading issue when using a Java EE container

### PR DESCRIPTION
**Pull request by vferries**
CLOSES : https://github.com/micrometer-metrics/micrometer/issues/2130

### CODE REVIEW

File: micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java [hunk 1]
```diff
@@ -204,7 +204,7 @@ private static boolean isManagementExtensionsPresent() {
 
         try {
             Class.forName("com.sun.management.GarbageCollectionNotificationInfo", false,
-                    JvmGcMetrics.class.getClassLoader());
+                    ManagementFactory.getMemoryMXBean().getClass().getClassLoader());
```
Comment (shakuzen) [inline micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java:207]: While this looks like it should work, the `MemoryMXBean` is not related to what this class is doing. Would something like `MemoryPoolMXBean.class.getClassLoader()` or even `ManagementFactory.class.getClassLoader()` work? Are those also loaded by the application container classloader?

Comment (shakuzen) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java:207]: I tested this with Wildfly and it looks like `MemoryPoolMXBean.class.getClassLoader()` resolves the classloader issue. Let me know if that's not the case for your application server.

Comment (vferries) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java:207]: I just checked it too and it works.

I will update the PR. You want it from 1.1.x branch to 1.1.x branch ?

**Comment by vferries** [comment]
PR rebased to 1.1.x, see here :
https://github.com/micrometer-metrics/micrometer/pull/2205

**Pull request closed without merging**